src/gtc/dace/expansion/utils.py

```diff
@@ -107,30 +107,45 @@ def mask_includes_inner_domain(mask: common.HorizontalMask):
 
 
 class HorizontalExecutionSplitter(eve.NodeTranslator):
+    @staticmethod
+    def is_horizontal_execution_splittable(he: oir.HorizontalExecution):
+        for stmt in he.body:
+            if isinstance(stmt, oir.HorizontalRestriction) and not mask_includes_inner_domain(
+                stmt.mask
+            ):
+                continue
+            elif isinstance(stmt, oir.LocalScalar):
+                continue
+            elif isinstance(stmt, oir.AssignStmt) and isinstance(stmt.left, oir.ScalarAccess):
+                continue
+            return False
```

Comment:
I don't understand why the check on line 117 is needed. `oir.HorizontalRestriction.body: List[oir.Stmt]` (is a list of OIR statements), which means that `stmt` can never be an `oir.LocalScalar`. Or am I misunderstanding something here?

Reply:
If you need a check that there are no local scalars defined on the horizontal execution, then something like `not he.declarations` should suffice.

Reply:
Oh yeah, I wrongly remembered that declarations were inlined at the HorizontalExecution level. Fixed.